### ISSUE TITLE
Issue with k8s.io/docs/tasks/configure-pod-container/pull-image-private-registry/

### ISSUE THREAD

**Issue by mabotn** [kubernetes/website#6793]
**This is a...** 

- [ ] Feature Request
- [x] Bug Report

**Problem:**

On my K8S cluster I had to use a private registry (Azure Container Registry) so I added a secret as mentioned using this command

`kubectl create secret docker-registry regsecret --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>`

but it appeared to create a secret that uses the type `kubernetes.io/dockercfg` which doesn't let kubernetes pull images from my private registry giving an error that includes  `unauthorized authentication required`

**Proposed Solution:**

After hours querying on Google and investigating with a friend @nzoueidi I found that the type should be `dockerconfigjson` so I created a new `secret.yml` file that has the following content

```
apiVersion: v1
kind: Secret
metadata:
  name: registry
data:
  .dockerconfigjson: eyJhdXRocyI6ey...m5KSyJ9fX0=
type: kubernetes.io/dockerconfigjson
```

After doing that, it appears that Kubernetes was able to pull the images from the registry

**Page to Update:**
https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

Kubernetes Version: 1.9
Additional Information: You may see ErrImagePull or ImagePullBackOff and if you hit describe you'll get unauthorized authentication required while pulling the image

**Comment by tengqm** [comment]
Thanks for opening this. Can you please open an issue at kubernetes/kubernetes project? It looks to me like a bug in the kubectl package.

**Comment by dims** [comment]
@tengqm @mabotn - known issue, fix has merged, 1.9.1 will have the fix - please see https://github.com/kubernetes/kubernetes/issues/45487#issuecomment-354751537 for more info

**Comment by tengqm** [comment]
Thanks, @dims 
That confirms the problem is not a doc issue.
